core/src/components/tab-button/tab-button.ionic.scss

```diff
@@ -86,3 +90,61 @@
 :host(.tab-button-shape-rectangular) {
   @include globals.border-radius(globals.$ion-border-radius-0);
 }
+
+// Tab Button: Badge
+// --------------------------------------------------
+
+:host ::slotted(ion-badge[vertical]:empty) {
+  @include padding(0, 0);
+
+  width: $tab-ionic-button-badge-size-emtpy;
+  height: $tab-ionic-button-badge-size-emtpy;
```

Comment:
Since the sizes change when in the tab button then we should add onto the sizes list for badge instead of modifying directly. Otherwise, we start contradicting the purpose of sizes. We should probably reach out to the design team to get clarification on this.

Reply:
Size related I implemented an overriding to set badge size with the same size independent of the given size attribute when inside tab button.